### PR TITLE
Next cylc nightly

### PR DESCRIPTION
**Pull request by james-bruten-mo**
I've had a request from Oliver asking if we're able to do some "next-version" testing of cylc with our rose-stems when it has an initial deploy in the near future. This PR adds that functionality to the nightly testing script. The script has been run with the example yaml file with a suitable example and produces the expected output. Unit tests have been added for the new function.

### CODE REVIEW

**Review by jennyhickson — COMMENTED** [review]
I feel like this would be cleaner and simpler to use the current "cylc_version" option that you've already got, rather than introducing a new "use_cylc_next" option? I don't see that 8-next is any different from 7 or 8 since they all get passed to "CYLC_VERSION=" in the same way... but maybe I've missed something.

**Comment by james-bruten-mo** [comment]
> I feel like this would be cleaner and simpler to use the current "cylc_version" option that you've already got, rather than introducing a new "use_cylc_next" option? I don't see that 8-next is any different from 7 or 8 since they all get passed to "CYLC_VERSION=" in the same way... but maybe I've missed something.

Thanks Jenny. The reason I didn't do that is the "cylc_version" option is used as a key to a dictionary (near the top of the file) so doing it that way would require some other modifications for that. Happy to do either - I suspect the changes would be a similar amount for both?

**Comment by jennyhickson** [comment]
> > I feel like this would be cleaner and simpler to use the current "cylc_version" option that you've already got, rather than introducing a new "use_cylc_next" option? I don't see that 8-next is any different from 7 or 8 since they all get passed to "CYLC_VERSION=" in the same way... but maybe I've missed something.
> 
> Thanks Jenny. The reason I didn't do that is the "cylc_version" option is used as a key to a dictionary (near the top of the file) so doing it that way would require some other modifications for that. Happy to do either - I suspect the changes would be a similar amount for both?

I can see that, but I think I'd still prefer to not add the extra option. You can probably use the first character of the cylc_version to determine 7 vs 8 for the key? or add an extra option to the dictionary that specifies the name etc for 8-next?

**Comment by james-bruten-mo** [comment]
Changed to just use the cylc_version option

**Review by jennyhickson — COMMENTED** [review]
Much prefer this, thanks. 

One last thought would be to make a small helper function to get the 7 or 8 from the cylc_version. Maybe something like:

def major_rev(cylc_version)
   return int(cylc_version[0])
   
thoughts?

**Comment by james-bruten-mo** [comment]
> Much prefer this, thanks.
> 
> One last thought would be to make a small helper function to get the 7 or 8 from the cylc_version. Maybe something like:
> 
> def major_rev(cylc_version) return int(cylc_version[0])
> 
> thoughts?

Sensible, done with some unit tests. Ta

**Review by jennyhickson — APPROVED** [review]
Thanks, that looks better.

File: nightly_testing/generate_test_suite_cron.py [hunk 1]
```diff
@@ -72,6 +74,16 @@ def run_command(command):
     )
 
 
+def major_cylc_version(cylc_version):
```
[the remark this answers is not in the document]
Comment (james-bruten-mo) [reply nightly_testing/generate_test_suite_cron.py:84]: No, more sensible to change the dict keys so done that. Cheers

File: nightly_testing/generate_test_suite_cron.py [hunk 2]
```diff
@@ -72,6 +74,16 @@ def run_command(command):
...
+    """
+    Return the major version of cylc being requested by cylc_version
+    Expected to be 7 or 8
+    """
+    if type(cylc_version) == int:
+        return cylc_version
+    return int(cylc_version[0])
```
[the remark this answers is not in the document]
Comment (james-bruten-mo) [reply nightly_testing/generate_test_suite_cron.py]: Done - used those delimiters which I think are fine to begin with

**Review by ericaneininger — APPROVED** [review]
Looks fine to me.  Thanks.

**Review by jennyhickson — APPROVED** [review]

**Pull request merged**